### PR TITLE
WT-12390 Convert checkpoint state to enum

### PR DESCRIPTION
**Pull request by bzm-jas**

### CODE REVIEW

File: src/include/checkpoint.h [hunk 1]
```diff
@@ -10,26 +10,29 @@
...
-#define WT_CHECKPOINT_STATE_COMMIT 11
-#define WT_CHECKPOINT_STATE_META_CKPT 12
-#define WT_CHECKPOINT_STATE_META_SYNC 13
-#define WT_CHECKPOINT_STATE_ROLLBACK 14
-#define WT_CHECKPOINT_STATE_LOG 15
-#define WT_CHECKPOINT_STATE_RUNNING 16
-#define WT_CHECKPOINT_STATE_ESTABLISH 17
-#define WT_CHECKPOINT_STATE_START_TXN 18
-#define WT_CHECKPOINT_STATE_CKPT_TREE 19
+/* Checkpoint state ordered by appearance in txn_ckpt.c. Inactive should always be 0. */
```
Comment (korteland) [inline src/include/checkpoint.h:13]: The first sentence makes it sound like the order matters - how about something more like `Inactive should always be 0. Other states happen to be roughly ordered by appearance in txn_ckpt.c.`

Comment (etienneptl) [reply src/include/checkpoint.h:13]: I don't think we should mention any file. Show we more say something about the checkpoint lifecyle?

File: src/include/checkpoint.h [hunk 2]
```diff
@@ -10,26 +10,29 @@
...
+enum Checkpoint_State {
```
Comment (etienneptl) [inline src/include/checkpoint.h:14]: I see in the WiredTiger code that we usually define enums following this syntax:
```
typedef enum {
...
} WT_CHECKPOINT_STATE;

File: src/include/checkpoint.h (hunk 1 above)
Comment (etienneptl) [reply src/include/checkpoint.h:13]: I like it. @korteland?

Comment (korteland) [reply src/include/checkpoint.h:13]: That's even better than what I suggested. Thanks!

File: src/include/checkpoint.h (hunk 2 above)
Comment (korteland) [reply src/include/checkpoint.h:14]: It's a mix. As far as I can tell, the pattern is that we use a `typedef` when we need to refer to it as a type. For a macro-ish use case, there's plenty of plain enums (c.f. `WT_CKPT_NONE` and `WT_CHILD_IGNORE`. So I think this declaration doesn't need a name at all :-)

File: src/include/checkpoint.h (hunk 1 above)
Comment (bzm-jas) [reply src/include/checkpoint.h]: comment updated here :[e5e713e](https://github.com/wiredtiger/wiredtiger/pull/10733/commits/e5e713e5402219959644f91e669bc0e2c9276cd8)

File: src/include/checkpoint.h (hunk 2 above)
Comment (bzm-jas) [reply src/include/checkpoint.h]: re-defined in: [e5e713e](https://github.com/wiredtiger/wiredtiger/pull/10733/commits/e5e713e5402219959644f91e669bc0e2c9276cd8)

Comment (korteland) [reply src/include/checkpoint.h]: Ah, anonymous enums are C++-only. Best to stick with your suggestion @etienneptl.

File: src/include/checkpoint.h [hunk 3]
```diff
@@ -10,26 +10,30 @@
...
-#define WT_CHECKPOINT_STATE_META_CKPT 12
-#define WT_CHECKPOINT_STATE_META_SYNC 13
-#define WT_CHECKPOINT_STATE_ROLLBACK 14
-#define WT_CHECKPOINT_STATE_LOG 15
-#define WT_CHECKPOINT_STATE_RUNNING 16
-#define WT_CHECKPOINT_STATE_ESTABLISH 17
-#define WT_CHECKPOINT_STATE_START_TXN 18
-#define WT_CHECKPOINT_STATE_CKPT_TREE 19
+/* Inactive should always be 0. Other states are roughly ordered by appearance in the checkpoint
+ * life cycle. */
```
Comment (etienneptl) [inline src/include/checkpoint.h:14]: Please format the comment correctly:

```suggestion
/* 
 * Inactive should always be 0. Other states are roughly ordered by appearance in the checkpoint
 * life cycle.
 */
```

**Review by etienneptl — APPROVED** [review]
LGTM % multiline comment format

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**